src/node_mime.cc

```diff
@@ -0,0 +1,370 @@
+          if (c == ';') {
+            break;
+          } else {
+            bool is_2_byte = (c & static_cast<unsigned char>(0xE0)) == 0xC0;
+            bool is_3_byte = (c & static_cast<unsigned char>(0xF0)) == 0xE0;
+            bool is_4_byte = (c & static_cast<unsigned char>(0xF8)) == 0xF0;
+            bool is_invalid = false;
+            if (is_2_byte) {
+              // need 2 bits of FF
+              if ((c & static_cast<unsigned char>(0x1C)) != 0) {
+                is_invalid = true;
+                right += 1;
+              } else {
+                right += 1;
+                if (right < src.size()) {
+                  c = src[right];
+                  if ((c & 0xC0) != 0x80) {
+                    is_invalid = true;
+                  }
+                } else {
+                  is_invalid = true;
+                }
+              }
+            } else if (is_3_byte) {
+              is_invalid = true;
+              right += 2;
+            } else if (is_4_byte) {
+              is_invalid = true;
+              right += 3;
+            } else if (!is_http_quoted_string_token(c)) {
+              is_invalid = true;
+            }
+            if (is_invalid && !saw_invalid_char) {
+              saw_invalid_value = true;
+              first_invalid_char = right;
+            }
+          }
+        }
+        {
+          size_t trim_right = right;
+          while (is_ascii_whitespace(src[trim_right - 1])) {
+            trim_right--;
+          }
+          if (!saw_invalid_char || first_invalid_char >= trim_right) {
+            parameter_value = src.substr(left, trim_right - left);
+          }
+        }
+      }
+    }
+    if (parameter_name.size() > 0 &&
+        parameter_value.size() > 0 &&
+        !saw_invalid_value) {
+      parameters_.push_back({ parameter_name, parameter_value });
+    }
+  }
+}
+
+
+void MIMEParser::Parse(const FunctionCallbackInfo<Value>& args) {
+  Environment* env = Environment::GetCurrent(args);
+  Isolate* isolate = args.GetIsolate();
+  HandleScope scope(isolate);
+
+  Local<Context> context = env->context();
+
+
+  if (args.IsConstructCall()) {
+    env->ThrowError("parse() must not be called using new");
+    return;
+  }
+
+  if (!args[0]->IsString()) {
+    env->ThrowError("first argument is not a string");
+    return;
+  }
+  if (!args[1]->IsObject()) {
+    env->ThrowError("second argument is not an object");
+    return;
+  }
+
+  Local<Object> dataSink = args[1].As<Object>();
+
+  Local<String> string = args[0].As<String>();
+  v8::String::Utf8Value source(isolate, string);
+
+  MIME mime(std::string(*source, source.length()));
+
+  if (mime.flags_ != mime_flags::MIME_FLAGS_NONE) {
+    env->ThrowError("Error parsing MIME");
+    return;
+  }
+
+  Local<String> type_key = FIXED_ONE_BYTE_STRING(isolate, "type");
+  Local<String> subtype_key = FIXED_ONE_BYTE_STRING(isolate, "subtype");
+  Local<String> parameters_key = FIXED_ONE_BYTE_STRING(isolate, "parameters");
```

Comment:
Put these on `Environment` so you don't have to recreate them every time.

Reply:
fixed